receiver/mongodbreceiver/ingegration_test.go

```diff
@@ -0,0 +1,154 @@
+		},
+		ExposedPorts: []string{"27017:27017"},
+		WaitingFor:   wait.ForListeningPort("27017").WithStartupTimeout(2 * time.Minute),
+	}
+)
+
+func TestMongodbIntegration(t *testing.T) {
+	t.Run("Running mongodb 4.0", func(t *testing.T) {
+		t.Parallel()
+		container := getContainer(t, containerRequest4_0)
+		defer func() {
+			require.NoError(t, container.Terminate(context.Background()))
+		}()
+		hostname, err := container.Host(context.Background())
+		require.NoError(t, err)
+
+		f := NewFactory()
+		cfg := f.CreateDefaultConfig().(*Config)
+		cfg.Hosts = []confignet.NetAddr{
+			{
+				Endpoint: net.JoinHostPort(hostname, "27017"),
+			},
+		}
+
+		consumer := new(consumertest.MetricsSink)
+		settings := componenttest.NewNopReceiverCreateSettings()
+		rcvr, err := f.CreateMetricsReceiver(context.Background(), settings, cfg, consumer)
+		require.NoError(t, err, "failed creating metrics receiver")
+
+		require.NoError(t, rcvr.Start(context.Background(), componenttest.NewNopHost()))
+		require.Eventuallyf(t, func() bool {
+			return len(consumer.AllMetrics()) > 0
+		}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
+		require.NoError(t, rcvr.Shutdown(context.Background()))
+
+		actualMtrics := consumer.AllMetrics()[0]
+
+		actualFile := filepath.Join("testdata", "integration", "actual.4_0.json")
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
+
+		expectedFile := filepath.Join("testdata", "integration", "expected.4_0.json")
+		expectedMetrics, err := golden.ReadMetrics(expectedFile)
+		require.NoError(t, err)
+
+		err = scrapertest.CompareMetrics(expectedMetrics, actualMtrics, scrapertest.IgnoreMetricValues())
+		require.NoError(t, err)
+	})
+	t.Run("Running mongodb 5.0", func(t *testing.T) {
+		t.Parallel()
+		container := getContainer(t, containerRequest5_0)
+		defer func() {
+			require.NoError(t, container.Terminate(context.Background()))
+		}()
+		hostname, err := container.Host(context.Background())
+		require.NoError(t, err)
+
+		f := NewFactory()
+		cfg := f.CreateDefaultConfig().(*Config)
+		cfg.Hosts = []confignet.NetAddr{
+			{
+				Endpoint: net.JoinHostPort(hostname, "27017"),
+			},
+		}
+
+		consumer := new(consumertest.MetricsSink)
+		settings := componenttest.NewNopReceiverCreateSettings()
+		rcvr, err := f.CreateMetricsReceiver(context.Background(), settings, cfg, consumer)
+		require.NoError(t, err, "failed creating metrics receiver")
+
+		require.NoError(t, rcvr.Start(context.Background(), componenttest.NewNopHost()))
+		require.Eventuallyf(t, func() bool {
+			return len(consumer.AllMetrics()) > 0
+		}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
+		require.NoError(t, rcvr.Shutdown(context.Background()))
+
+		actualMtrics := consumer.AllMetrics()[0]
+
+		actualFile := filepath.Join("testdata", "integration", "actual.5_0.json")
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
```

Comment:
This should be considered temporary code.
```suggestion
		actualFile := filepath.Join("testdata", "integration", "expected.5_0.json")
		err = golden.WriteMetrics(actualFile, actualMtrics)
		require.NoError(t, err)
```

Run this one time and then remove all of these lines. Then you have your "expected" file.

Reply:
ok, I have known how to do next.